lib/extractor/layer.ts

```diff
@@ -135,6 +201,48 @@ function getLayersContentAndArchiveManifest(
   };
 }
 
+function getOciLayersContentAndArchiveManifest(
+  imageIndex: OciImageIndex,
+  manifestCollection: OciArchiveManifest[],
+  layers: Record<string, ExtractedLayers>,
+): { layers: ExtractedLayers[]; manifest: OciArchiveManifest } {
+  // filter empty layers
+  // get the layers content without the name
+  // reverse layers order from last to first
+
+  // get manifest file first
+  const manifest = getOciManifest(imageIndex, manifestCollection);
+  const filteredLayers = manifest.layers
+    .filter((layer) => Object.keys(layers[layer.digest]).length !== 0)
+    .map((layer) => layers[layer.digest])
+    .reverse();
```

Comment:
Do we know if the layers are in order of how they appeared in the "Dockerfile"?

Reply:
I think so from reading the docs: "The array MUST have the base layer at index 0. Subsequent layers MUST then follow in stack order"